packages/@lwc/synthetic-shadow/src/3rdparty/polymer/path-composer.ts

```diff
@@ -20,22 +20,34 @@ import { getOwnerDocument } from '../../shared/utils';
 import { isInstanceOfNativeShadowRoot } from '../../env/shadow-root';
 import { SyntheticShadowRoot } from '../../faux-shadow/shadow-root';
 
+interface Slotable extends Node {
+    assignedSlot: HTMLSlotElement;
+}
+
 export function pathComposer(startNode: EventTarget, composed: boolean): EventTarget[] {
     const composedPath: EventTarget[] = [];
-    let current: EventTarget | null = startNode;
-    const startRoot: Window | Node =
-        startNode instanceof Window ? startNode : (startNode as Node).getRootNode();
+
+    let startRoot: Window | Node;
+    if (startNode instanceof Window) {
+        startRoot = startNode;
+    } else if (startNode instanceof Node) {
+        startRoot = startNode.getRootNode();
+    } else {
+        return composedPath;
+    }
+
+    let current: Window | Node | null = startNode;
     while (!isNull(current)) {
         composedPath.push(current);
 
-        if (current instanceof Element) {
-            // Text also implements the Slottable mixin but is not an EventTarget, nor a ParentNode,
-            // which means it is safe to only look for Elements here.
-            const assignedSlot: HTMLSlotElement | null = current.assignedSlot;
+        const _slotable: unknown = current;
```

Comment:
I don't think you need all of this. You can keep the `current instanceof Element` check:

```ts
        if (current instanceof Element) {
            // Text also implements the Slottable mixin but is not an EventTarget, nor a ParentNode,
            // which means it is safe to only look for Elements here.
            const assignedSlot: HTMLSlotElement | null = current.assignedSlot;
            if (!isNull(assignedSlot)) {
                current = assignedSlot;
            } else {
                current = current.parentNode;
            }
```

The `: HTMLSlotElement | null` part is still needed here for some reason. It appears that tsc is not able to downcast the `HTMLSlotElement` to `Window | Node | null`.

Reply:
The current logic seems to be making an invalid assumption because `Text` does include the `EventTarget` interface.